src/App.tsx

```diff
@@ -1,61 +1,117 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
+
+const getUserById = (userId: number): User | null => {
+  const correctUser = usersFromServer.find(user => user.id === userId);
+
+  return correctUser || null;
+};
+
+export const preparedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState<Todo[]>(preparedTodos);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [hasSelectedError, setHasSelectedError] = useState(false);
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setHasSelectedError(!selectedUserId);
+    setHasTitleError(!title);
+
+    const getMaxId = Math.max(...todos.map(todo => todo.id)) + 1;
+
+    const clearForm = () => {
+      setTitle('');
+      setSelectedUserId(0);
+    };
+
+    const newTodo = {
+      id: getMaxId,
+      title,
+      completed: false,
+      userId: +selectedUserId,
```

Comment:
do we need to convert the value to the number here? you're already stored the number in the state